day_two_guide/topics/node_maintenance.adoc

```diff
@@ -0,0 +1,338 @@
+
+If using `EmptyDir`, the user can use the node filesystem to store temporary
+data but if the node needs to be evacuated or if the node fails, the data
+is lost. The location of the data in the *{rhocp}* node is located in
+`/var/lib/origin/openshift.local.volumes` and the temporary storage size can
+be configured:
+
+NOTE: If the XFS filesystem hosting that folder is mounted with the
+`gquota` option in the `/etc/fstab`
+* the matching security context contraint’s `fsGroup` type set to `MustRunAs`
+* The `/etc/origin/node/node-config.yml` is configured as:
+
+[subs=+quotes]
+----
+volumeConfig:
+  localQuota:
+     perFSGroup: *512Mi*
+----
+
+If there are pods using local storage and the evacuation needs to be
+performed, the admin user can use the `--delete-local-data` flag in the
+`oc adm drain` command to force the pod evacuation even if the local data
+is lost.
+
+WARNING: Use `EmptyDir` volumes for temporary and non-important data only.
+
+=== Pod Disruption Budget
+
+`PodDisruptionBudget` is supported starting with {product-title} version 3.6. `PodDisruptionBudget` is an object that specifies the minimum percentage of
+pod replicas or number of pods that must be running at a time on voluntary
+disruptions such as evacuate a node.
+
+For example, in a situation where the `PodDisruptionBudget` is configured to
+have a minimum of two pods running and there are three replicas, but two of
+those replicas are running in the same node. Without the `PodDisruptionBudget`,
+both replicas are evicted at the same time, but using the `PodDisruptionBudget`
+the evacuation procedure respect it and the pods are evicted respecting one by
+one.
+
+NOTE: This object is useful to avoid the application to fail in node maintenance
+tasks or cluster upgrade procedures and the value depends on the application
+but it shouldn't be the same number as the `ReplicationController` otherwise
+the node cannot be evacuated normally and the evacuation needs to be forced.
+
+To create a `PodDisruptionBudget`:
+
+. Create the `guestbook-pdb.yaml`:
++
+----
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: guestbook-pdb
+spec:
+  selector:
+    matchLabels:
+      app: guestbook
+  minAvailable: 2 <1>
+EOF
+----
+<1> Specify a minimum of two pods. To specify a percentage the percentage (`%`) symbol is used as: `80%`.
+
+. Create the `PodDisruptionBudget`:
++
+----
+$ oc create -f guestbook-pdb.yaml
+----
+
+. Verify it:
++
+----
+$ oc get pdb
+NAME            MIN-AVAILABLE   ALLOWED-DISRUPTIONS   AGE
+guestbook-pdb   2               1                     6s
+----
++
+In this example the "guestbook" application has three replicas so one pod is
+allowed to be "disrupted".
+
+[WARNING]
+====
+`PodDisruptionBudget` objects cannot be edited. In the event of any changes then
+it is required to deploy a new `PodDisruptionBudget` object should be created.
```

Comment:
either 'it is required to deploy' or 'should be created' - both are not needed and confuse things.